src/materials/czm/PureElasticCZM.C

```diff
@@ -0,0 +1,114 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "InterfaceValueTools.h"
+#include "PureElasticCZM.h"
+
+registerMooseObject("DeerApp", PureElasticCZM);
+
+template <> InputParameters validParams<PureElasticCZM>() {
+  InputParameters params = validParams<CZMMaterialBase>();
+  params.addClassDescription(
+      "time dependent interface damage model with linear interpolation");
+  params.addRequiredParam<Real>("E", "interface normal elastic moduls");
+  params.addRequiredParam<Real>("G", "interface shear elastic moduls");
+  params.addRequiredParam<Real>("interface_thickness",
+                                "initial interface thickness");
+  params.addParam<Real>(
+      "penetration_penalty", 1e2,
+      "the penetration penalty applied after (duN < -interface_thickness)");
+  return params;
+}
+
+PureElasticCZM::PureElasticCZM(const InputParameters &parameters)
+    : CZMMaterialBase(parameters), _E(getParam<Real>("E")),
+      _G(getParam<Real>("G")),
+      _interface_thickness(getParam<Real>("interface_thickness")),
+      _penetration_penalty(getParam<Real>("penetration_penalty"))
+
+{}
+
+RealVectorValue PureElasticCZM::computeTraction() {
+  // The convention for ordering the traction is N, T, S, where N is the normal
+  // direction, and T and S are two arbitrary tangential directions.
+  RealVectorValue traction;
+
+  ComputeNormalTraction(traction);
+  ComputeShearTraction(traction);
+
+  return traction;
+}
+
+RankTwoTensor PureElasticCZM::computeTractionDerivatives() {
+  RankTwoTensor traction_jump_derivatives;
+
+  ComputeNormalTractionDerivatives(traction_jump_derivatives);
+  ComputeShearTractionDerivatives(traction_jump_derivatives);
+
+  return traction_jump_derivatives;
+}
+
+void PureElasticCZM::ComputeShearTraction(RealVectorValue &traction) {
+  Real C = ComputeShearStiffness();
+  for (unsigned int i = 1; i < 3; i++)
```

Comment:
Here and elsewhere: this should not be a magic "3".  What happens in 2D?  Or at the very least you should be throwing an error for 2D meshes.

See the moose tensor_mechanics kernels for how they deal with  it.

Reply:
This is a magic 3 :grin: as the interface model always work in 3D even for 2D meshes.
I'll add a comment but is also already documented in MOOSE in the CZMMaterialBase class